File: android/app/src/main/java/com/phoneagent/core/product/ProductCandidate.kt
package com.phoneagent.core.product

/**
 * Phone Agent - Step 2K Product Candidate Model (Android)
 */
data class ProductCandidate(
    val candidateId: String,
    val title: String,
    val visiblePrice: Double? = null,
    val priceQualifier: String? = null,
    val currency: String? = null,
    val rating: Double? = null,
    val reviewCount: Int? = null,
    val thumbnailUri: String? = null,
    val productUrl: String? = null,
    val sourcePosition: Int,
    val sourceFingerprint: String,
    val confidence: Double = 1.0,
    val rawTextSnippet: String? = null,
    val isSponsored: Boolean = false
)

data class ProductResearchRequest(
    val requestId: String,
    val query: String,
    val category: String? = null,
    val minPrice: Double? = null,
    val maxPrice: Double? = null,
    val currency: String? = "USD",
    val desiredAttributes: List<String> = emptyList(),
    val maximumCandidates: Int = 5,
    val createdAt: Long = System.currentTimeMillis()
)

File: src/core/product/ProductCandidate.ts
/**
 * Phone Agent - Step 2K Product Candidate Model
 * Represents a visible product candidate discovered on the Amazon search results screen.
 * Strictly source-backed from accessibility UI nodes; zero fabricated metrics.
 */

export interface ProductCandidate {
  candidateId: string;
  title: string;
  visiblePrice?: number;
  priceQualifier?: string; // e.g. "with coupon", "after discount", "starting at"
  currency?: string;
  rating?: number;
  reviewCount?: number;
  thumbnailUri?: string; // Local media URI only (e.g. content://, file://)
  productUrl?: string;
  sourcePosition: number;
  sourceFingerprint: string;
  confidence: number;
  rawTextSnippet?: string;
  isSponsored?: boolean;
}

export interface ProductResearchRequest {
  requestId: string;
  query: string;
  category?: string;
  priceRange?: {
    min?: number;
    max?: number;
    currency?: string;
  };
  desiredAttributes?: string[];
  maximumCandidates?: number;
  createdAt: number;
}
